File: Lisharefrontend/src/modules/social/friend/services/friendService.js
import axiosInstance from "/src/modules/platform/api/axiosInstance";
import { ENDPOINTS } from "/src/modules/platform/api/endpoints";

export const friendService = {
  getFriends: () => axiosInstance.get(ENDPOINTS.friend.all),
  getFriendsForUser: (userId) => axiosInstance.get(ENDPOINTS.friend.allForUser(userId)),
  getPending: () => axiosInstance.get(ENDPOINTS.friend.pending),
  getSent: () => axiosInstance.get(ENDPOINTS.friend.sent),
  request: (userId) => axiosInstance.post(ENDPOINTS.friend.request(userId)),
  accept: (userId) => axiosInstance.post(ENDPOINTS.friend.accept(userId)),
  reject: (userId) => axiosInstance.post(ENDPOINTS.friend.reject(userId)),
  cancel: (userId) => axiosInstance.delete(ENDPOINTS.friend.cancel(userId)),
  unfriend: (userId) => axiosInstance.delete(ENDPOINTS.friend.unfriend(userId))
};

File: Lisharebackend/src/main/java/com/socialApp/Lishare/modules/social/friend/service/FriendService.java
package com.socialApp.Lishare.modules.social.friend.service;

import com.socialApp.Lishare.modules.platform.user.entity.User;
import com.socialApp.Lishare.modules.social.friend.dto.FriendActionResponse;
import jakarta.transaction.Transactional;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface FriendService {

    // Send a friend request
    FriendActionResponse sendFriendRequest(Long senderId, Long receiverId);

    // Accept a friend request
    FriendActionResponse acceptFriendRequest(Long senderId, Long receiverId);

    // Reject a friend request
    FriendActionResponse rejectFriendRequest(Long senderId, Long receiverId);

    // Remove a friend (unfriend)
    FriendActionResponse unfriend(Long user1Id, Long user2Id);

    // Get all accepted friends of a user
    List<User> getFriends(Long userId);

    // Get all pending friend requests for a user
    List<User> getPendingRequests(Long userId);

    // Get all outgoing pending friend requests for a user
    List<User> getSentRequests(Long userId);



}

